lib/active_model/serializer/adapter.rb

```diff
@@ -11,6 +12,47 @@ class Adapter
 
       attr_reader :serializer
 
+      # Only the Adapter class has these methods.
+      # None of the sublasses have them.
+      class << ActiveModel::Serializer::Adapter
+        def adapter_map
+          ADAPTER_MAP
+        end
+
+        def register(name, klass)
+          adapter_map.update(name.to_s.underscore => klass)
+        end
+
+        def get(adapter)
```

Comment:
> 
I'll take a look when I'm next at a computer.  Do you like the new interface and behavior?  Or have an alternative in mind?
>